catalog/spatial/wfs/spatial-wfs-featuretransformer/src/main/java/org/codice/ddf/spatial/ogc/wfs/featuretransformer/impl/WfsRouteBuilder.java

```diff
@@ -33,22 +33,33 @@
 
   @Override
   public void configure() {
+    /*
+     * This route is designed to be called via a Camel proxy using the FeatureTransformationService
+     * interface. Arguments are bound to the message body as an array, i.e.
+     *
+     * method: apply(InputStream featureCollection, WfsMetadata metadata)
+     * message body: new Object[] { featureCollection, metadata }
+     */
     from(FEATURECOLLECTION_ENDPOINT_URL)
         .id("TransformFeatureCollectionRoute")
+        .streamCaching()
         .setHeader("metadata", simple("${body[1]}"))
-        .setHeader("xml", simple("${body[0]}"))
+        // Stream caching doesn't work when the input stream is located in a header. Copying the
+        // stream to the body first is a workaround. It causes the stream to be cached, then we can
+        // copy that cache to the header so we can access it later.
+        .setBody(simple("${body[0]}"))
```

Comment:
I suppose the body was already in memory, but is there anyway that this could blow up memory, thus creating a possible DNS attack?

Reply:
If the stream is too large, Camel won't load it into memory. From the docs [1]:
> large stream messages (over 64 KB in Camel 2.11 or older, and 128 KB from Camel 2.12 onwards) will be cached in a temporary file instead — Camel itself will handle deleting the temporary file once the cached stream is no longer necessary.

[1] https://camel.apache.org/manual/latest/stream-caching.html